core/templates/domain/topic/topic-update.service.ts

```diff
@@ -62,6 +69,35 @@ export class TopicUpdateService {
     changeDict.cmd = command;
     let changeObj = new Change(changeDict, apply, reverse);
     this.undoRedoService.applyChange(changeObj, entity);
+    this._updateTopicEditorBrowserTabUnsavedChangesStatus(entity);
+  }
+
+  private _updateTopicEditorBrowserTabUnsavedChangesStatus(entity) {
+    let topicId: string = null;
+    try {
+      // If the entity is a SubtopicPage, then the corresponding topic
+      // id is given by getTopicId().
+      topicId = entity.getTopicId();
+    } catch (e) {
+      // Otherwise, the topic id can be accessed via entity.getId().
+      topicId = entity.getId();
+    }
```

Comment:
Maybe better to explicitly check the kind of entity and get the corresponding id. Alternatively, you can create an interface called "EntityEditorPage" that declares a "getId" method that the other "XYZPage" classes can implement. That way you can just call "getId" on the entity (you can also use that as the type definition for "entity" argument in this method.)

Reply:
I have used the first alternative now. Thanks!